flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -21,18 +26,98 @@ class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
         options.logger(SentryLevel.debug,
             'Scheduler binding is null. Can\'t auto detect app start time.');
       } else {
-        schedulerBinding.addPostFrameCallback((timeStamp) {
+        schedulerBinding.addPostFrameCallback((timeStamp) async {
           // ignore: invalid_use_of_internal_member
-          _native.appStartEnd = options.clock();
+          final appStartEnd = options.clock();
+          _native.appStartEnd = appStartEnd;
+
+          if (!_native.didFetchAppStart) {
+            final nativeAppStart = await _native.fetchNativeAppStart();
+            final measurement = nativeAppStart?.toMeasurement(appStartEnd!);
+
+            // We filter out app start more than 60s.
+            // This could be due to many different reasons.
+            // If you do the manual init and init the SDK too late and it does not
+            // compute the app start end in the very first Screen.
+            // If the process starts but the App isn't in the foreground.
+            // If the system forked the process earlier to accelerate the app start.
+            // And some unknown reasons that could not be reproduced.
+            // We've seen app starts with hours, days and even months.
+            if (nativeAppStart == null ||
+                measurement == null ||
+                measurement.value >= 60000) {
+              _appStartTracker?.setAppStartInfo(null);
+              return;
+            }
+
+            final appStartInfo = AppStartInfo(
+              DateTime.fromMillisecondsSinceEpoch(
+                  nativeAppStart.appStartTime.toInt()),
+              appStartEnd,
+              measurement,
+            );
+
+            _appStartTracker?.setAppStartInfo(appStartInfo);
+          } else {
+            _appStartTracker?.setAppStartInfo(null);
+          }
         });
       }
     }
 
-    options.addEventProcessor(NativeAppStartEventProcessor(_native));
+    options.addEventProcessor(NativeAppStartEventProcessor(appStartTracker: _appStartTracker));
 
     options.sdk.addIntegration('nativeAppStartIntegration');
   }
 }
 
 /// Used to provide scheduler binding at call time.
 typedef SchedulerBindingProvider = SchedulerBinding? Function();
+
+@internal
+class AppStartInfo {
```

Comment:
`l`: You could change this to an inner class of `AppStartTracker` as this class only makes sense in combination with the `AppStartTracker`. That's a good use case of an inner class.

Reply:
unfortunately inner classes are not really a thing in Dart so not sure if we can meaningfully combine those two